src/status_im/protocol/handlers.cljs

```diff
@@ -1,291 +1,204 @@
-                                  :removed-from-at timestamp
-                                  :is-active       false}])))))
-
-(register-handler :participant-left-group
-  (u/side-effect!
-    (fn [{:keys [current-public-key]}
-         [_ {:keys                                    [from]
-             {:keys [group-id timestamp] :as payload} :payload}]]
-      (when (and (not= current-public-key from)
-                 (chats/is-active? group-id)
-                 (> timestamp (chats/get-property group-id :timestamp)))
-        (participant-left-group-message group-id from payload)
-        (dispatch [::remove-identity-from-chat group-id from])
-        (dispatch [::remove-identity-from-chat! group-id from])))))
-
-(register-handler ::remove-identity-from-chat
-  (fn [db [_ chat-id id]]
-    (update-in db [:chats chat-id :contacts]
-               #(remove (fn [{:keys [identity]}]
-                          (= identity id)) %))))
-
-(register-handler ::remove-identity-from-chat!
-  (u/side-effect!
-    (fn [_ [_ group-id identity]]
-      (chats/remove-contacts group-id [identity]))))
-
-(register-handler :participant-invited-to-group
-  (u/side-effect!
-    (fn [{:keys [current-public-key chats]}
-         [_ {:keys                                            [from]
-             {:keys [group-id identity message-id timestamp]} :payload}]]
-      (let [admin (get-in chats [group-id :group-admin])]
-        (when (= from admin)
-          (dispatch
-            [:received-message
-             (participant-invited-to-group-message group-id current-public-key identity from message-id timestamp)])
-          (when-not (= current-public-key identity)
-            (dispatch [:add-contact-to-group! group-id identity])))))))
-
-(register-handler :add-contact-to-group!
-  (u/side-effect!
-    (fn [_ [_ group-id identity]]
-      (when-not (chats/has-contact? group-id identity)
-        (dispatch [::add-contact group-id identity])
-        (dispatch [::store-contact! group-id identity])))))
-
-(register-handler ::add-contact
-  (fn [db [_ group-id identity]]
-    (update-in db [:chats group-id :contacts] conj {:identity identity})))
-
-(register-handler ::store-contact!
-  (u/side-effect!
-    (fn [_ [_ group-id identity]]
-      (chats/add-contacts group-id [identity]))))
-
-(defn save-message-status! [status]
-  (fn [_ [_
-          {:keys                                        [from]
-           {:keys [message-id ack-of-message group-id]} :payload}]]
+(re-frame/reg-fx
+  ::you-removed-from-group-message
+  (fn [{:keys [from message-id timestamp group-id]}]
+    (let [remover-name (:name (contacts/get-by-id from))
+          message  (->> [(or remover-name from) (i18n/label :t/removed-from-chat)]
+                        (string/join " ")
+                        (system-message message-id timestamp))
+          message' (assoc message :group-id group-id)]
+      (re-frame/dispatch [:received-message message']))))
```

Comment:
Same as above re why is this even an effect

Reply:
same,it reads data from data-store